### ISSUE TITLE
Raw data doc ids should be required to start with sourceId

### ISSUE THREAD

**Issue by jo** [eHealthAfrica/kazana#117]

**Comment by gr2m** [comment]
a use case that we have to keep in mind is: a user who uploads a report for say Oct 2015 might realise afterwards that in fact that was the report for Nov 2015, so she/he might want to change it, which would change the source id and therefor the raw-data doc._id. So internally, that would be a deleted and re-create, which would then trigger a cleanup and re-creation of all indicators extracted from that report, which makes sense, as the indicators have the source id stored as well

**Comment by jo** [comment]
But from what I understand `sourceId` is the [transformer key](https://github.com/eHealthAfrica/kazana-server/blob/master/lib/compose-plugin.js#L180), not the docid --- not?

**Comment by gr2m** [comment]
yes that's correct. I just commented because the issue says that we should require the `doc._id`s in `raw-data` to start with the `sourceId`s

**Comment by jo** [comment]
Currently the integrated data doc ids are in the form
```
<rawDataDoc.id>:<server.app.utils.uuid>
```

This ticket is meant to add the `sourceId`:
```
<sourceId>:<rawDataDoc.id>:<server.app.utils.uuid>
```

**Comment by gr2m** [comment]
yeah, I think we agree, I was wondering if `<rawDataDoc.id>` should byitself already be prefixed by `sourceId` or not
